grpc/codec.go

```diff
@@ -0,0 +1,41 @@
+package grpc
+
+import (
+	"io"
+
+	"google.golang.org/grpc/encoding"
+)
+
+// CodecName is the name registered for the proto compressor.
+const CodecName = "FiberCodec"
+
+// FiberCodec is a custom codec to prevent marshaling and unmarshalling
+// when unnecessary, base on the inputs
+type FiberCodec struct{}
+
+// Marshal will attempt to pass the request directly if it is a byte slice,
+// otherwise unmarshal the request proto using the default implementation
+func (FiberCodec) Marshal(v interface{}) ([]byte, error) {
+	b, ok := v.([]byte)
+	if ok {
+		return b, nil
+	}
+	defaultCodec := encoding.GetCodec("proto")
+	return defaultCodec.Marshal(v)
+}
+
+// Unmarshal will attempt to write the request directly if it is a writer,
+// otherwise unmarshal the request proto using the default implementation
+func (FiberCodec) Unmarshal(data []byte, v interface{}) error {
+	myType, ok := v.(io.Writer)
```

Comment:
What's the reason it's expecting `io.Writer` instead of `[]byte` similar to `Marshall` ?

Reply:
using `[]byte`, the callers data doesn't get manipulated (if v is set to data). 

It will reset to its original value in this [block](https://github.com/grpc/grpc-go/blob/master/rpc_util.go#L760). Not sure if im missing something straight forward.

Reply:
ah ok got it

Reply:
i used `[]byte` initially and it didnt work, I always thought `interface{}` var is a pointer but doesn't seems to behave that way.  😅